### DISCUSSION TITLE
[Request] Add issue drafts and autosaving

### DISCUSSION THREAD

**Discussion by Errorbot1122** [community/community#7801] [Projects and Issues]
I know that Github already saves the **title** of an issue if you are using a templet but I want it to save the whole thing, not just the title

I suggest you use **cookies** or some type of offline saving method, so if your internet goes out or something happens, you can still save your draft, you just can't submit it until it can connect again.

**Comment by janbrasna** [comment]
there was some sort of autosaving before, using local storage i guess… that slowly disappeared during the past few years, maybe because of browser support, limiting the data persistence maybe? dunno. but i still remember the good ol days when i knew i won't lose any content in github textboxes — nowadays i'm sure that after a browser crash, restart, os update or anything that happens while i'm afk, i'll find an empty page :(

all text typed into issues, comments, prs etc. needs more love, and saving it a bit more thoroughly…

**Reply by Errorbot1122** [reply]
you could just use cookies, if they could store enough data

**Comment by rwspielman** [comment]
This has killed me in the past couple of days. accidentally erased a large issue I was writing by clicking edit labels. Can we perhaps at least have a browser warning about leaving the page will erase changes to prevent accidentally leaving?

**Reply by janbrasna** [reply]
Well, the sad thing is that this functionality used to work in the past decade so I usually rely on this heavily. Therefore I usually end up pretty bummed finding out that I've lost maybe hours worth of preparation in wip texts when this caching fails after reload/navigation :(

**Comment by julie777** [comment]
I just accidentally click on a button while creating and issue. I had been working on it for about 30 minutes. 
It should either be saved as a draft, which is good for many reasons,
or in this particular case there should be a notice about leaving the page to make sure you know you are throwing away your work. Many web pages do this, so it isn't hard.

**Comment by julie777** [comment]
In addition, most web forms will still have the information if you use the back button. Please implement this. Going back took me back to the Issues page and not to the new issue that I was working on.
